### ISSUE TITLE
Error when using blockmeter in a single world

### ISSUE THREAD

**Issue by horseloyal** [ModProg/BlockMeter#60]
The start selection works, trying to do the other one is when things go wrong.
game version 1.21, fabric 0.15.11, fabric-api-0.100.3+1.21

![image](https://github.com/ModProg/BlockMeter/assets/171438743/6ee7212e-ff14-4ecb-99ee-6faf2f87ecf9)

**Comment by ModProg** [comment]
Does this happen without any other mods enabled?

Can you share the logs when this happens?

**Comment by horseloyal** [comment]
Crashes even with just blockmeter and fabric api, but the mod works perfectly fine on a paper server.

[latest.log](https://github.com/user-attachments/files/16046533/latest.log)

**Comment by babybonobo** [comment]
I am having this exact same issue although a few times it worked without crashing, usually on the 2nd selection (to begin measuring) I get kicked from my singleplayer world with that error.  Interestingly afterward the game places me into the multiplayer server list menu after I get kicked.

**Comment by ModProg** [comment]
As a workaround, it might help to disable sending the boxes (the settings menu can be accessed via [Mod Menu](https://modrinth.com/mod/modmenu))
![grafik](https://github.com/ModProg/BlockMeter/assets/11978847/0bd2b7d6-648f-4b55-a681-6765d539c638)


Though, I'm very confused by the error message. As I don't seem to be able to reproduce it.

Searching for this error, all I could find was https://www.reddit.com/r/MinecraftHelp/comments/1dlccqg/singleplayer_issue_internal_exception/

I added some print statements in the package decoding, maybe it tells something, could use run this version and share the logs:
[blockmeter-1.14-1.21.zip](https://github.com/user-attachments/files/16057407/blockmeter-1.14-1.21.zip)

**Comment by babybonobo** [comment]
It's weird, I can't get it to crash any more.  It was doing it in a certain measurement during a mob grinder I was building over the ocean, but now when I try it around my base it's not crashing so I wasn't able to provide you with a log of it crashing again.  Thanks for the workaround though, I will use it if I experience more issues and maybe provide you with a log otherwise.

**Comment by ModProg** [comment]
Yeah, sadly the log didn't really provide any useful information yet, I'll try to think of more things I could try to log.

**Comment by maybeToby90** [comment]
> As a workaround, it might help to disable sending the boxes (the settings menu can be accessed via [Mod Menu](https://modrinth.com/mod/modmenu)) ![grafik](https://private-user-images.githubusercontent.com/11978847/344766841-0bd2b7d6-648f-4b55-a681-6765d539c638.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI5MzE2OTMsIm5iZiI6MTcyMjkzMTM5MywicGF0aCI6Ii8xMTk3ODg0Ny8zNDQ3NjY4NDEtMGJkMmI3ZDYtNjQ4Zi00YjU1LWE2ODEtNjc2NWQ1MzljNjM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA2VDA4MDMxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdiN2Y5ZDU3MDg4N2Q0ZjFjZWE5YzIwYzQ3NjJhZDIzMTQ3MTdlZDRjMTIxYmM2NzYwZTkzNmMwZGMwMWUyY2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Bpz96SY8U2BBE51S6Smkp39AvxK6_TBtGO7udYzZPKE)
> 
> Though, I'm very confused by the error message. As I don't seem to be able to reproduce it.
> 
> Searching for this error, all I could find was https://www.reddit.com/r/MinecraftHelp/comments/1dlccqg/singleplayer_issue_internal_exception/
> 
> I added some print statements in the package decoding, maybe it tells something, could you maybe run this version and share the logs: [blockmeter-1.14-1.21.zip](https://github.com/user-attachments/files/16057407/blockmeter-1.14-1.21.zip)

omg, thanks so much, this worked for me. and i think i dont need this option :) thank you! ❤️

**Comment by Alchny** [comment]
Извините за русский. Проблема не появляется на сервере с ядром paper но в одиночном плоском мире она есть.
[crash-2024-09-23_13.32.47-
[2024-09-23-2.log](https://github.com/user-attachments/files/17096408/2024-09-23-2.log)
client.txt](https://github.com/user-attachments/files/17096398/crash-2024-09-23_13.32.47-client.txt)
